### ISSUE TITLE
[FEATURE REQUEST] Add consistent behavior for failed integrations in fullchem_mod.F90

### ISSUE THREAD

**Issue by yantosca** [geoschem/geos-chem#1606]
I've noticed some inconsistent behavior (and possibly an error) for the block where the integration fails in fullchem_mod.F90:

```F90
       !=====================================================================
       ! Try another time if it failed
       !=====================================================================
       IF ( IERR < 0 ) THEN

#if defined( MODEL_GEOS ) || defined( MODEL_WRF )
          ! Save a copy of the C vector (GEOS and WRF only)
          localC    = C
#endif

          ! Reset first time step and start concentrations
          ! Retry the integration with non-optimized settings
          RCNTRL(3) = 0.0_dp
          C         = 0.0_dp

          ! Disable auto-reduce solver for the second iteration for safety
          IF ( Input_Opt%Use_AutoReduce ) THEN
             RCNTRL(12) = -1.0_dp ! without using ICNTRL
          ENDIF

          ! Update rates again
          CALL Update_RCONST( )

          ! Start timer
          IF ( Input_Opt%useTimers ) THEN
             CALL Timer_Start( TimerName = "     Integrate 2",               &
                               InLoop    =  .TRUE.,                          &
                               ThreadNum = Thread,                           &
                               RC        = RC                               )
          ENDIF

          ! Integrate again
          CALL Integrate( TIN,    TOUT,    ICNTRL,                           &
                          RCNTRL, ISTATUS, RSTATE, IERR                     )

          ! Stop timer
          IF ( Input_Opt%useTimers ) THEN
             CALL Timer_End( TimerName = "     Integrate 2",                 &
                             InLoop    =  .TRUE.,                            &
                             ThreadNum = Thread,                             &
                             RC        = RC                                 )
          ENDIF
```
In particular:
1. For GEOS and WRF, a `localC` array is defined to store the values of C (but not for CLASSIC or GCHP)
2. The `C` array is zeroed out instead of being reset to pre-chemistry concentrations.'

I would recommend adding this change:

```F90
       !=====================================================================
       ! Initialize the KPP "C" vector of species concentrations [molec/cm3]
       !=====================================================================
       DO N = 1, NSPEC
          SpcID = State_Chm%Map_KppSpc(N)
          C(N)  = 0.0_dp
          IF ( SpcId > 0 ) C(N) = State_Chm%Species(SpcID)%Conc(I,J,L)
       ENDDO
    
       ! Store pre-chemistry values in a vector
       C_pre_chem = C

       ...

       IF ( IERR < 0 ) THEN
   

          ! Reset first time step and start concentrations
          ! Retry the integration with non-optimized settings
          RCNTRL(3) = 0.0_dp
   
          ! Reset concentrations to pre-chemistry values
          C         = C_pre_Chem

          ...
```
Thoughts?  Would this mess anything up for GEOS or WRF?

@msulprizio @lizziel @Jourdan-He @SaptSinha @christophkeller @jimmielin

**Comment by jimmielin** [comment]
Hi Bob, WRF-GC can follow similar behavior as GEOS. It would be best to reset the concentrations to pre-chemistry values and retry the integration.

I think GEOS and WRF-GC additionally has some extra behavior to skip the integration if it still fails and increment a counter (`KppError`?) somewhere down the line. It would be best to keep that too for WRF-GC.

Copying @fengx7 as well. Thanks!
Haipeng

**Comment by msulprizio** [comment]
The corresponding PR for this feature request has been merged into the 14.2.0 development branch.
